### PR TITLE
WIP add support for loss function masking

### PR DESCRIPTION
**Pull request by HastingsGreer**

### CODE REVIEW

**Comment by neel-dey** [comment]
hey @HastingsGreer I saw this in my feed and was curious how *ICON decided to implement masks. I wanted to chime in on one thing:

```
loss = loss * mask
return torch.mean(loss)
```

should probably be 

```
loss = loss * mask
return torch.sum(loss) / torch.sum(mask)  # assuming that the mask is binary foreground
```

otherwise the loss is dominated by locations zeroed out by the masks. Now if you're using Adam, this shouldn't matter for direct pairwise optimization but I've empirically seen it impact network training.

(sorry to butt in uninvited!)
